packages/mobile/src/paymentRequest/utils.ts

```diff
@@ -1,50 +1,52 @@
-import { PaymentRequest } from 'src/account/types'
 import { AddressToE164NumberType, SecureSendPhoneNumberMapping } from 'src/identity/reducer'
 import { getAddressValidationType } from 'src/identity/secureSend'
 import { AddressValidationCheckCache } from 'src/paymentRequest/IncomingPaymentRequestListScreen'
+import { PaymentRequest } from 'src/paymentRequest/types'
 import { NumberToRecipient, Recipient, RecipientKind } from 'src/recipients/recipient'
 
-export function getRecipientFromPaymentRequest(
+// Returns a recipient for the SENDER of a payment request
+// i.e. this account when outgoing, another when incoming
+export function getRequesterFromPaymentRequest(
   paymentRequest: PaymentRequest,
+  addressToE164Number: AddressToE164NumberType,
```

Comment:
This will only work if the request recipient has the request sender in their address book and has sent to them before, right?

Reply:
Correct. I don't think we can avoid that requirement unfortunately